src/main/java/net/neoforged/neoforge/common/extensions/IPlayerExtension.java

```diff
@@ -92,4 +98,33 @@ default boolean isCloseEnough(Entity entity, double dist) {
         return aabb.distanceToSqr(eye) < dist * dist;
     }
 
+    /**
+     * Request to open a GUI on the client, from the server
+     * <p>
+     * Refer to {@link MenuType#create(IContainerFactory)} for how to provide a function to consume
+     * these GUI requests on the client.
+     *
+     * @param menuProvider A supplier of container properties including the registry name of the container
+     * @param pos          A block pos, which will be encoded into the additional data for this request
+     *
+     */
+    default OptionalInt openMenu(MenuProvider menuProvider, BlockPos pos) {
+        return openMenu(menuProvider, buf -> buf.writeBlockPos(pos));
+    }
+
+    /**
+     * Request to open a GUI on the client, from the server
+     * <p>
+     * Refer to {@link MenuType#create(IContainerFactory)} for how to provide a function to consume
+     * these GUI requests on the client.
+     * <p>
+     * The maximum size for #extraDataWriter is 32600 bytes.
+     *
+     * @param menuProvider    A supplier of container properties including the registry name of the container
+     * @param extraDataWriter Consumer to write any additional data the GUI needs
+     * @return The window ID of the opened GUI, or empty if the GUI could not be opened
+     */
+    default OptionalInt openMenu(MenuProvider menuProvider, Consumer<FriendlyByteBuf> extraDataWriter) {
```

Comment:
This and the other `openMenu` overload should have the `@Nullable` annotation on the `MenuProvider` as the only implementation has it so we may as well make it more explicitly clear that modders can pass a null provider to it.